ddtrace/tracer/spancontext.go

```diff
@@ -23,9 +23,10 @@ var _ ddtrace.SpanContext = (*spanContext)(nil)
 type spanContext struct {
 	// the below group should propagate only locally
 
-	trace *trace // reference to the trace that this span belongs too
-	span  *span  // reference to the span that hosts this context
-	drop  bool   // when true, the span will not be sent to the agent
+	trace  *trace // reference to the trace that this span belongs too
+	span   *span  // reference to the span that hosts this context
+	drop   bool   // when true, the span will not be sent to the agent
+	errors int64  // number of spans with errors in this trace
```

Comment:
Do we use the number of errors per trace somewhere? I thought we were only doing number of errors per span

Reply:
One span can have only one error. However, a trace can have multiple spans with errors, and we count that here. We count errors at the trace level because we want to know if a span belongs to a trace which has caused an error, so that it can be sampled for stats. Is this the wrong approach? I assumed we will count all spans that belong to a trace which has any error. Or, is the right approach to count only the specific spans of a trace that have the actual error and ignore the rest?

I might have misunderstood here.

Reply:
Ah, I didn't realize it was for sampling (and not for stats). Sounds good!